internal/models/subscription.go

```diff
@@ -5,16 +5,17 @@ import (
 	"time"
 
 	"github.com/google/uuid"
+	"gorm.io/datatypes"
 	"gorm.io/gorm"
 )
 
 type Subscription struct {
-	Id        string    `gorm:"primaryKey;type:varchar(255)" json:"id"`
-	WebhookId string    `json:"webhook_id" gorm:"type:varchar(255)"`
-	EventType string    `json:"event_type"`
-	FilterCriteria map[string]interface{} `gorm:"column:filter_criteria" json:"filter_criteria"`
-	CreatedAt time.Time `json:"created_at"`
-	UpdatedAt time.Time `json:"updated_at"`
+	Id             string         `gorm:"primaryKey;type:varchar(255)" json:"id"`
+	WebhookId      string         `json:"webhook_id" gorm:"type:varchar(255)"`
+	EventType      string         `json:"event_type"`
+	FilterCriteria datatypes.JSON `json:"filter_criteria"`
```

Comment:
from the prd, it was a json type, i didnt understand the `map[string]inteface{}` ... but with this style, if we want to use filterCriteria in our code.

```import "encoding/json"

// When setting the filter criteria
filterMap := map[string]interface{}{
    "sender": "0xabc",
    "amount": 100,
}
jsonData, err := json.Marshal(filterMap)
if err != nil {
    // handle error
}

subscription := models.Subscription{
    WebhookId:      "some-webhook-id",
    EventType:      "transfer",
    FilterCriteria: datatypes.JSON(jsonData),
}

// if we want to read it back to map
var filterMap map[string]interface{}
err := json.Unmarshal(subscription.FilterCriteria, &filterMap)
if err != nil {
    // handle error
}
```